www/js/main.js

```diff
@@ -0,0 +1,76 @@
+/* global horsey */
+
+// Requires jQuery, Horsey, and configuration of the below config object.
+
+(function () {
+
+  'use strict';
+
+  var config = {
+    // HTML element to attach this to
+    element: 'input.entry',
+    // backend route from which we get user objects (to get usernames for the suggestions)
+    route: '/user/search',
+    // attribute of query object sent to backend with request
+    reqAtt: 'pattern',
+    // attribute we expect each object to have in the response
+    // (expecting a JSON array of objects)
+    resAtt: '_id'
+  };
+  var DELKEY = 'U+0008';
+  $(config.element).on('keyup', function () {
+    var $eventTarget = $(event.target);
+    var spaceCheck = $eventTarget.val().split('@')[0];
+    if (spaceCheck[spaceCheck.length - 1] === ' ' || spaceCheck === '') {
+
+      var str = $eventTarget.val().split('@')[1];
+      if (str && str.length >= 2 && event.keyIdentifier !== DELKEY) {
+        if (!config.horse) {
+          getAutoData(str, function (formattedData) {
+            setHorsey(formattedData);
+          });
+        } else {
+          config.horse.show();
+        }
+      }
+      if (event.keyIdentifier === DELKEY && ((str && str.length <= 1) || str === undefined)) {
```

Comment:
Does that imply a module to be imported, maybe a class since I need to preserve state of what key was pressed/etc. (depending on how I wind up handling that?)